hack/generator/spec/versioning.md

```diff
@@ -0,0 +1,192 @@
+
+#### Resolution
+
+TBC
+
+### Do we merge all properties from all versions of an API? [Resolved]
+
+There are two potential approaches to take when defining the storage version of an object. We can merge just the properties defined on the versions of the API for which we are emitting support, or we can merge all the properties defined across all versions of the API.
+
+#### Discussion
+
+If we only merge properties from versions of the versions of the API for which we are emitting support, we run the risk of the storage API becoming incompatible with the course of time. To illustrate, consider the following definition for a `Person`:
+
+``` go
+package v20180101
+
+type Person struct {
+    FirstName string
+    LastName  string
+}
+```
+
+For the v20200101 version of the API, with different fields that work cleanly across a wider set of cultures:
+
+``` go
+package v20200101
+
+type Person struct {
+    FullName   string
+    FamilyName string
+    KnownAs    string
+    SortKey    string
+}
+```
+
+Assuming we are generating support for the most recent two versions of the API, we end up with a storage schema that looks like this:
+
+``` go
+package v1
+
+type Person struct {
+    FirstName string
+    LastName  string
+    FullName   string
+    FamilyName string
+    KnownAs    string
+    SortKey    string
+}
+```
+
+However, when a new version of the API is introduced, say v20210101 merging only the most recent two versions would give us the following storage schema:
+
+``` go
+package v1
+
+type Person struct {
+    FullName   string
+    FamilyName string
+    KnownAs    string
+    SortKey    string
+}
+```
+
+This would be incompatible with any existing versions in storage, breaking the backward compatibility promise we should be adherent to.
+
+The problem occurs in different forms regardless of the number of supported versions being retained. A threshold of two versions was used above to keep the example constrained.
+
+#### Resolution
+
+The storage version will be generated by combining all available versions of the API to create a superset, regardless of the number of versions being emitted by the code generator.
+
+### How do we handle extensibility?
+
+How should we accomodate situations where the generated code needs to be augmented with some hand coded additions?
+
+#### Discussion
+
+We'll already be generating the required `ConvertTo()` and `ConvertFrom()` methods defined by the `Convertible` interface, as a part of supporting the usual hub-spoke type conversion model.
+
+One possible approach would be to generate a custom interface for each type, with a name based on the ARM API being targetted
+
+``` go 
+package v20200101
+
+type AssignablePerson {}
+    AssignTo(person v1.Person) error 
```

Comment:
minor: assuming `v1.Person` is a struct this should be a ptr

Reply:
Fixed.